iocore/cache/Cache.cc

```diff
@@ -3444,7 +3444,7 @@ HTTPInfo_v21::copy_and_upgrade_unmarshalled_to_v23(char *&dst, char *&src, size_
   hdr_size = ROUND(s_hdr->unmarshal_size(), HDR_PTR_SIZE);
   if (hdr_size > length)
     return false;
-  memcpy(d_hdr, s_hdr, hdr_size);
+  memcpy(static_cast<void *>(d_hdr), s_hdr, hdr_size);
```

Comment:
Why not *d_hdr = *s_hdr ?

Reply:
Maybe we can do that later. The same code is on master too. I kept the change minimum.

Reply:
I don't understand your reasoning.  The general rule is,  don't bypass strong type checking unless you really need to.  I don't think it makes sense to break this rule to be consistent with other code that's breaking this rule.  It seems to just increase the backlog for refactoring.